features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesView.kt

```diff
@@ -402,15 +409,48 @@ private fun MessagesViewTopBar(
         },
         actions = {
             if (inRoomCallsEnabled) {
-                IconButton(onClick = onJoinCallClicked) {
-                    Icon(CommonDrawables.ic_compound_video_call, contentDescription = null) // TODO add proper content description once we have the state
+                if (isCallOngoing) {
+                    JoinCallMenuItem(onJoinCallClicked = onJoinCallClicked)
+                } else {
+                    IconButton(onClick = onJoinCallClicked) {
+                        Icon(CommonDrawables.ic_compound_video_call, contentDescription = null) // TODO add proper content description once we have the state
+                    }
                 }
             }
+            Spacer(Modifier.width(8.dp))
         },
         windowInsets = WindowInsets(0.dp)
     )
 }
 
+@Composable
+private fun JoinCallMenuItem(
+    modifier: Modifier = Modifier,
+    onJoinCallClicked: () -> Unit,
+) {
+    Material3Button(
+        onClick = onJoinCallClicked,
+        colors = ButtonDefaults.buttonColors(
+            contentColor = ElementTheme.colors.bgCanvasDefault,
+            containerColor = ElementTheme.colors.iconAccentTertiary
+        ),
+        contentPadding = PaddingValues(horizontal = 10.dp, vertical = 0.dp),
+        modifier = modifier.heightIn(min = 36.dp),
+    ) {
+        Icon(
+            modifier = Modifier.size(20.dp),
+            resourceId = CommonDrawables.ic_compound_video_call,
+            contentDescription = stringResource(CommonStrings.action_join)
```

Comment:
A11y: Since there is a text in the button, with the same string, I think the contentDescription can be null here.